src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -61,10 +63,13 @@ internal class CompletionSource : ForegroundThreadAffinitizedObject, IAsyncCompl
         private readonly ITextView _textView;
         private readonly bool _isDebuggerTextView;
         private readonly ImmutableHashSet<string> _roles;
+        private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _streamingPresenters;
```

Comment:
this being an IEnumerable bugs me a bit (i'm always worried about unintentional multiplicative costs).  Can we just realize it into an ImmutableArray?

Reply:
It will go away when https://github.com/dotnet/roslyn/pull/34117 is merged.